src/Integration/WPCOMProxy.php

```diff
@@ -0,0 +1,215 @@
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Registerable;
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use WP_REST_Response;
+use WP_REST_Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WPCOMProxy
+ *
+ * Initializes the hooks to filter the data sent to the WPCOM proxy depending on the query parameter gla_syncable.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Integration
+ */
+class WPCOMProxy implements Service, Registerable {
+
+	/**
+	 * The Post types to be filtered.
+	 *
+	 * @var array
+	 */
+	public static $post_types_to_filter = [
+		'product'           => [
+			'meta_query' => [
+				[
+					'key'     => '_wc_gla_visibility',
+					'value'   => 'sync-and-show',
+					'compare' => '=',
+				],
+			],
+		],
+		'shop_coupon'       => [
+			'meta_query' => [
+				[
+					'key'     => '_wc_gla_visibility',
+					'value'   => 'sync-and-show',
+					'compare' => '=',
+				],
+				[
+					'key'     => 'customer_email',
+					'compare' => 'NOT EXISTS',
+				],
+			],
+		],
+		'product_variation' => [
+			'meta_query' => null,
+		],
+	];
+
+	/**
+	 * Register all filters.
+	 */
+	public function register(): void {
+		// Allow to filter by gla_syncable.
+		add_filter(
+			'woocommerce_rest_query_vars',
+			function ( $valid_vars ) {
+				$valid_vars[] = 'gla_syncable';
+				return $valid_vars;
+			}
+		);
+
+		foreach ( array_keys( self::$post_types_to_filter ) as $object_type ) {
+			$this->register_object_types_filter( $object_type );
+		}
+	}
+
+	/**
+	 * Register the filters for a specific object type.
+	 *
+	 * @param string $object_type The object type.
+	 */
+	protected function register_object_types_filter( string $object_type ): void {
+		add_filter(
+			'woocommerce_rest_prepare_' . $object_type . '_object',
+			[ $this, 'check_item_is_syncable' ],
+			9,
+			3
+		);
+
+		add_filter(
+			'woocommerce_rest_prepare_' . $object_type . '_object',
+			[ $this, 'filter_metadata' ],
+			10,
+			3
+		);
+
+		add_filter(
+			'woocommerce_rest_' . $object_type . '_object_query',
+			[ $this, 'filter_by_metaquery' ],
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Whether the data should be filtered.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return bool
+	 */
+	protected function should_filter_data( WP_REST_Request $request ): bool {
+		// WPCOM proxy will set the gla_syncable to 1 if the request is coming from the proxy and it is the Google App.
+		return $request->get_param( 'gla_syncable' ) === '1';
+	}
+
+	/**
+	 * Check if a single item is syncable.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param mixed            $item     The item.
+	 * @param WP_REST_Request  $request  The request object.
+	 *
+	 * @return WP_REST_Response The response object updated.
+	 */
+	public function check_item_is_syncable( $response, $item, WP_REST_Request $request ): WP_REST_Response {
```

Comment:
More than just a check (boolean return) I find this more as a filtering of the response. 

Maybe "filter_response_by_syncable_item" ?? Just an idea, the point is that check asses a true/false statement